pkg/controllers/sync/dispatch/managed.go

```diff
@@ -55,24 +51,16 @@ const IndexRolloutPlans = "federation_placement_rollout"
 // interface required for dispatching operations to managed resources.
 type FederatedResourceForDispatch interface {
 	TargetName() common.QualifiedName
-	TargetKind() string
 	TargetGVK() schema.GroupVersionKind
+	TargetGVR() schema.GroupVersionResource
 	TypeConfig() *fedcorev1a1.FederatedTypeConfig
-	// Replicas is the number of replicas specified in the federated object.
-	Replicas() (*int64, error)
 	// Object returns the federated object.
-	Object() *unstructured.Unstructured
+	Object() fedcorev1a1.GenericFederatedObject
 	VersionForCluster(clusterName string) (string, error)
 	ObjectForCluster(clusterName string) (*unstructured.Unstructured, error)
-	ApplyOverrides(
-		obj *unstructured.Unstructured,
-		clusterName string,
-		otherOverrides fedtypesv1a1.OverridePatches,
-	) error
+	ApplyOverrides(obj *unstructured.Unstructured, clusterName string) error
 	RecordError(errorCode string, err error)
```

Comment:
Can we merge `RecordError` and `RecordEvent` since all `RecordError` is doing now is to emit an event?

Reply:
`RecordError` records an event with event type set as `EventTypeWarning`.